codegen/smithy-dafny-codegen-cli/src/main/java/software/amazon/polymorph/CodegenCli.java

```diff
@@ -119,6 +120,11 @@ private static Options getCliOptions() {
             .desc("<optional> AWS SDK for Java version to use: v1, or v2 (default)")
             .hasArg()
             .build())
+          .addOption(Option.builder()
+            .longOpt("dafny-version")
+            .desc("<optional> Dafny version to use. Defaults to 4.1.0")
```

Comment:
You sure about that? I feel like defaulting to 4.1 is something we may regret

Reply:
I was only trying to be backwards compatible for anyone else using the CLI. Note that the smithy plugin avoids this by incrementing the latest edition and making it required!

Reply:
After offline conversation, elected to make it required in the CLI as well.